src/main/java/com/google/devtools/build/lib/analysis/config/CoreOptions.java

```diff
@@ -984,7 +984,7 @@ public FragmentOptions getExec() {
     exec.affectedByStarlarkTransition = affectedByStarlarkTransition;
     exec.outputDirectoryNamingScheme = outputDirectoryNamingScheme;
     exec.compilationMode = hostCompilationMode;
-    exec.isExec = false;
```

Comment:
This looks like a bug from when we migrated `getHost()` -> `getExec()` and dropped the host config from calling this method. I think with this change we no longer need https://github.com/bazelbuild/bazel/blob/eb4a78c4caa4532236b0f8f0e37b3ce157467d41/src/main/java/com/google/devtools/build/lib/analysis/config/ExecutionTransitionFactory.java#L128?

Reply:
Yes, I delete that line in this PR.

Reply:
Oh sorry - I missed that part of the diff.

Reply:
> By "previous" do you mean `--experimental_output_directory_naming_scheme=legacy` and `--experimental_exec_configuration_distinguisher=legacy`? (trying to remember which combination of flags we expect when!)

Yes, exactly, although I don't think the value of `--experimental_exec_configuration_distinguisher=legacy` matters. The `affected_by_starlark_transition` logic was resistant against top-level flags changing that were never transitioned on, whereas without this change transition in the exec cfg can be sensitive to such changes.